File: task-03/Elixir/diamond_file.ex
defmodule LoopModule do
  def upper_half(_star, 0) do
  end
  def upper_half(star, num) do
    star_new = String.slice(star, 1..-1)
    IO.write(output, star_new)
    star_new = star_new <> " *"
    upper_half(star_new, num-1)
  end
  def lower_half(_star, 0) do
  end
  def lower_half(star, num) do
    IO.write(output, star_new)
    star_new = String.slice(star, 0..-3)
    star_new = " " <> star_new
    lower_half(star_new, num-1)
  end
end

{:ok, num} = File.read("input.txt")
{:ok, output} = File.open("output.txt", [ :write])
star = String.duplicate(" ", num)
LoopModule.upper_half(star, num)
lower = String.duplicate("* ", num)
LoopModule.lower_half(lower, num)
File.close(output)

File: task-03/Python/diamond_file.py
string = open("input.txt", "r")
f = open("output.txt", "w")
num = int(string.read())
star = " "*(num)
for x in range(num):
  star = star[1:]
  star += " *"
  f.write(star + "\n")
for i in range(1,num+1):
  star = star[:-2]
  star = " " + star
  f.write(star + "\n")